Update core doc (#13324)
openvinotoolkit/openvino: src/core/docs/api_details.md

# OpenVINO Core API

OpenVINO Core API contains two folders:

>  * [ngraph](../include/ngraph/) - is a legacy API, this API is not developed more. Only aliases to new operation and operation sets extend this API.

Reviewer: is not developed more ---> is no longer being developed
maybe "aliases to new operationS"?

>  * [openvino](../include/openvino/) - current public API, this part will be described below.

Reviewer: ```suggestion
 * [openvino](../include/openvino/) - current public API, this part is described below.
```

## Structure of Core API
<pre>
 <code>
 <a href="../include/openvino">openvino/</a>                  // Common folder with OpenVINO 2.0 API
    <a href="../include/openvino/core/">core/</a>                   // Contains common classes which are responsible for model representation
    <a href="../include/openvino/op/">op/</a>                     // Contains all supported OpenVINO operations
    <a href="../include/openvino/opsets/">opsets/</a>                 // Contains definitions of each official OpenVINO opset
    <a href="../include/openvino/pass/">pass/</a>                   // Defines classes for developing transformation and several common transformations
    <a href="../include/openvino/runtime/">runtime/</a>                // Contains OpenVINO tensor definition
 </code>
</pre>

## Main structures for model representation

> * `ov::Model` is located in [openvino/core/model.hpp](../include/openvino/core/model.hpp) and provide API for model representation. For more details please read [OpenVINO Model Representation Guide](https://docs.openvino.ai/latest/openvino_docs_OV_UG_Model_Representation.html).

Reviewer: ```suggestion
* `ov::Model` is located in [openvino/core/model.hpp](../include/openvino/core/model.hpp) and provides API for model representation. For more details, read [OpenVINO Model Representation Guide](https://docs.openvino.ai/latest/openvino_docs_OV_UG_Model_Representation.html).
```

* `ov::Node` is a base class for all OpenVINO operations, the class is located in the [openvino/core/node.hpp](../include/openvino/core/node.hpp).

> * `ov::Shape` and `ov::PartialShape` classes represents shapes in the OpenVINO, these classes are located in the [openvino/core/shape.hpp](../include/openvino/core/shape.hpp) and [openvino/core/partial_shape.hpp](../include/openvino/core/partial_shape.hpp) respectively. For more information please read [OpenVINO Shapes representation](./shape_propagation.md#openvino-shapes-representation).

Reviewer: ```suggestion
* `ov::Shape` and `ov::PartialShape` classes represent shapes in the OpenVINO, these classes are located in the [openvino/core/shape.hpp](../include/openvino/core/shape.hpp) and [openvino/core/partial_shape.hpp](../include/openvino/core/partial_shape.hpp) respectively. For more information, read [OpenVINO Shapes representation](./shape_propagation.md#openvino-shapes-representation).
```

* `ov::element::Type` class represents element type for OpenVINO Tensors and Operations. The class is located in the [openvino/core/type/element_type.hpp](../include/openvino/core/type/element_type.hpp).

> * `ov::Tensor` is used for memory representation inside the OpenVINO. The class is located in the [openvino/runtime/tensor.hpp](../include/openvino/runtime/tensor.hpp).

Reviewer: ```suggestion
* `ov::Tensor` is used for memory representation inside OpenVINO. The class is located in the [openvino/runtime/tensor.hpp](../include/openvino/runtime/tensor.hpp).
```
